File: assets/a2.js
class Solution {
  recSubsetsWithDup(nums, i, curr, ans) {
     if (i >= nums.length) {
      const c = [...curr];
      ans.push(c);
      return
    }
    curr.push(nums[i]);
    this.recSubsetsWithDup(nums, i + 1, curr, ans);
    curr.pop();
    if (i < nums.length && ans[i] != ans[i + 1]) {
        this.recSubsetsWithDup(nums, i + 1, curr, ans);
    } else {
        while (i < nums.length && nums[i] == nums[i + 1]) {
          i++;
        }
    }
  }
  subsetsWithDup(nums) {
    //your code goes here
    let ans = [];
    this.recSubsetsWithDup(nums, 0, [], ans);
    console.log(ans)
    return ans;
  }
}
const a = new Solution();
a.subsetsWithDup([1, 2, 2]);

File: assets/a2.cpp
class Solution{
    public:
    bool checkSum(vector<int> & curr , int k ){
        int sum = 0 ; 
        for(int i = 0 ; i < curr.size() ; i++){
            if(sum == k) return true;

        }
        return false;
    }
    bool recCheckS(vector<int> & nums , int k , int i , vector <int> & curr){
        if(i == nums.size()){
            return checkSum(curr , k);
        }
        curr.push_back(nums[i]);
        
       bool res =   recCheckS(nums , k , i +1 , curr );
               curr.pop_back();

       bool res1 = res ;
       if(res == false){
         res1 =  recCheckS(nums , k , i , curr);
       }
       return res || res1  ;
       
    }
    bool checkSubsequenceSum(vector<int>& nums, int k) {
         //your code goes here
        vector <int> curr ;
       return  recCheckS(nums , k , 0 , curr  );
    }
};
#include <vector>
 #include <iostream>
using namespace std;
  
int main() {
    Solution solution;
    vector<int> nums = {1, 2, 3};  
    int k = 6;
    bool result = solution.checkSubsequenceSum(nums , k);
    // cout << "the " << result  ;

    // Print the result
     
    return 0;
}